src/Client.php

```diff
@@ -140,15 +148,15 @@ public function setRawDataStream($data, $enctype = null)
      */
     public function setMethod($method = HttpRequest::METHOD_GET)
     {
-        if ($method == HttpRequest::METHOD_GET) {
+        if ($method === HttpRequest::METHOD_GET) {
             $this->setRequestMethod(HttpRequest::METHOD_GET);
-        } elseif ($method == HttpRequest::METHOD_POST) {
+        } elseif ($method === HttpRequest::METHOD_POST) {
             $this->setRequestMethod(HttpRequest::METHOD_POST);
-        } elseif ($method == HttpRequest::METHOD_PUT) {
+        } elseif ($method === HttpRequest::METHOD_PUT) {
             $this->setRequestMethod(HttpRequest::METHOD_PUT);
-        } elseif ($method == HttpRequest::METHOD_DELETE) {
+        } elseif ($method === HttpRequest::METHOD_DELETE) {
             $this->setRequestMethod(HttpRequest::METHOD_DELETE);
-        } elseif ($method == HttpRequest::METHOD_HEAD) {
+        } elseif ($method === HttpRequest::METHOD_HEAD) {
             $this->setRequestMethod(HttpRequest::METHOD_HEAD);
```

Comment:
Prior to the strict equality checks, you should `strtoupper` or `strtolower` (Whichever matches the constant case) the `$method` argument